submission_offsets.py

```diff
@@ -0,0 +1,68 @@
+from flask.cli import FlaskGroup
+import click
+from collections import defaultdict
+
+import db
+import webserver
+
+from sqlalchemy import text
+
+cli = FlaskGroup(add_default_commands=False, create_app=webserver.create_app_flaskgroup)
+
+@cli.command(name='add-offsets')
+@click.option("--limit", "-l", default=10000)
+def add_offsets(limit):
+	"""Update lowlevel submission offsets with a specified limit."""
+	incremental_add_offset(limit)
+
+def incremental_add_offset(limit):
+	with db.engine.connect() as connection:
+		# Find the next batch of items to update
+		batch_query = text("""
+			SELECT id, gid
+			  FROM lowlevel
+			 WHERE submission_offset IS NULL
```

Comment:
this query will need an order by id, to ensure that we're getting items in submission order

Reply:
Does this query order by id when no other order is specified? I was trying this query using an order by id, and trying it without and it was returning the same order (by id) in both cases.

Reply:
the sql standard doesn't specify a behaviour if you don't specify `order by`. perhaps by a programming decision or luck postgres appears to return items in id order, but this is not guaranteed. We should always be safe and explicitly set it.

Reply:
Okay thanks, I've specified order by id in the latest commit.